subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -60,157 +60,157 @@
         StartParameterBuildOptions.options = Collections.unmodifiableList(options);
     }
 
-    public static List<BuildOption<StartParameter>> get() {
+    public static List<BuildOption<StartParameterInternal>> get() {
         return options;
     }
 
     private StartParameterBuildOptions() {
     }
 
-    public static class ProjectCacheDirOption extends StringBuildOption<StartParameter> {
+    public static class ProjectCacheDirOption extends StringBuildOption<StartParameterInternal> {
         public ProjectCacheDirOption() {
             super(null, CommandLineOptionConfiguration.create("project-cache-dir", "Specify the project-specific cache directory. Defaults to .gradle in the root project directory."));
         }
 
         @Override
-        public void applyTo(String value, StartParameter settings, Origin origin) {
+        public void applyTo(String value, StartParameterInternal settings, Origin origin) {
             Transformer<File, String> resolver = new BasicFileResolver(settings.getCurrentDir());
             settings.setProjectCacheDir(resolver.transform(value));
         }
     }
 
-    public static class RerunTasksOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
+    public static class RerunTasksOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
         public RerunTasksOption() {
             super(null, CommandLineOptionConfiguration.create("rerun-tasks", "Ignore previously cached task results."));
         }
 
         @Override
-        public void applyTo(StartParameter settings, Origin origin) {
+        public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setRerunTasks(true);
         }
     }
 
-    public static class RecompileScriptsOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
-        public static final String DEPRECATION_MESSAGE = "Support for --recompile-scripts was deprecated and is scheduled to be removed in Gradle 5.0.";
+    public static class RecompileScriptsOption extends EnabledOnlyBooleanBuildOption<StartParameterInternal> {
         public RecompileScriptsOption() {
-            super(null, CommandLineOptionConfiguration.create("recompile-scripts", "Force build script recompiling.").deprecated(DEPRECATION_MESSAGE));
+            super(null, CommandLineOptionConfiguration.create("recompile-scripts", "Force build script recompiling.").deprecated());
```

Comment:
:o: Maybe break out the command line parameter as constant so it can be reused. In `appyTo` we could just use `settings.addDeprecation("--" + LONG_OPTION);`